internal/datastore/filetracker/filetracker.go

```diff
@@ -59,13 +74,32 @@ func (ft FileTracker) Exist(file string) bool {
 		return false
 	}
 
+	fileInfo, err := os.Stat(file)
+	if err != nil {
+		ft.logger.Debugf("Error retrieving file info for '%s' (%s).", file, err)
+		return false
+	}
+
+	if item.ModTime.Equal(fileInfo.ModTime()) {
+		ft.logger.Debugf("File modification time has not changed for '%s'.", file)
+		return true
+	}
+
 	hash, err := ft.Hasher.Hash(file)
 	if err != nil {
 		return false
 	}
 
 	// checks if the file is the same (equal value)
-	if item.Hash() == hash {
+	if item.Hash == hash {
+		ft.logger.Debugf("File hash has not changed for '%s'.", file)
+
+		// updates file marker with mtime to speed up comparison on next run
+		item.ModTime = fileInfo.ModTime()
+		if err = ft.repo.Put(file, item); err != nil {
```

Comment:
I think that `Exist` is idempotent in the sense that it always returns the same output, no matter how many times you call it. But it's true that maybe is not obeying all the SOLID principles. 

For me it's not something to address now, but maybe readability could be improved extracting part to another method/function.